### PR TITLE
V2 of NGD 

### PR DESCRIPTION
**Pull request by manekinekko**
Here is the new mono-repo structure:

```
src/modules
├── cli
│   ├── bin
│   ├── dist
│   ├── node_modules
│   ├── package.json
│   ├── src
│   ├── test
│   └── tsconfig.json
├── compiler
│   ├── dist
│   ├── node_modules
│   ├── package.json
│   ├── src
│   ├── test
│   └── tsconfig.json
├── core
│   ├── dist
│   ├── node_modules
│   ├── package.json
│   ├── src
│   ├── test
│   ├── tsconfig.json
│   └── yarn.lock
└── transformer
    ├── dist
    ├── node_modules
    ├── package.json
    ├── src
    ├── test
    └── tsconfig.json
```

The new package are:
- @compodoc/ngd-cli
- @compodoc/ngd-core
- @compodoc/ngd-compiler
- @compodoc/ngd-transformer

Using the API:

```javascript
import { DotEngine } from '@compodoc/ngd-transformer';
import { Compiler } from '@compodoc/ngd-compiler';

//...

let compiler = new Compiler(
      files, {
        tsconfigDirectory: cwd
      }
    );

let deps = compiler.getDependencies();

let engine = new DotEngine({
      output: program.output,
      displayLegend: program.displayLegend,
      outputFormats: program.outputFormats.split(',')
    });
    engine
      .generateGraph(deps)
      .then(file => { ... });
```

### CODE REVIEW

File: src/modules/transformer/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    "contributors": [
+        "vogloblinsky"
+    ],
+    "repository": {
+        "type": "git",
+        "url": "https://github.com/compodoc/ngd.git"
+    },
+    "private": false,
+    "license": "MIT",
+    "dependencies": {
```
Comment (vogloblinsky) [inline src/modules/transformer/package.json]: dot, fs-extra and q dependencies missing ?

File: src/modules/compiler/dist/compiler.js [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        return this.getSymbolDeps(props, 'directives').map(function (name) {
+            var identifier = _this.parseDeepIndentifier(name);
+            identifier.selector = _this.findComponentSelectorByName(name);
+            identifier.label = '';
+            return identifier;
+        });
+    };
+    Compiler.prototype.parseDeepIndentifier = function (name) {
+        var nsModule = name.split('.');
+        console.log(nsModule);
```
Comment (vogloblinsky) [inline src/modules/compiler/dist/compiler.js]: for debugging ? still in output logs ;)

**Comment by vogloblinsky** [comment]
Using https://github.com/compodoc/compodoc-demo-todomvc-angular2 demo project.
Linking to required scoped modules with `ln -s`
I started just a simple Node.js script : 

`
let ngdT = require('@compodoc/ngd-transformer');
let ngdC = require('@compodoc/ngd-compiler');

let files = [ '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/about/about-routing.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/about/about.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/about/about.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/about/compodoc/compodoc.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/about/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/about/todomvc/todomvc.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/app-routing.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/app.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/app.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/footer/footer.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/footer/footer.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/footer/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/header/header.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/header/header.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/header/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/home/home-routing.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/home/home.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/home/home.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/home/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/list/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/list/list.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/list/list.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/list/todo/index.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/list/todo/todo.component.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/list/todo/todo.module.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/shared/directives/do-nothing.directive.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/shared/interfaces/interfaces.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/shared/models/todo.model.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/shared/pipes/first-upper.pipe.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/shared/services/emitter.service.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/app/shared/services/todo.store.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/environments/environment.prod.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/environments/environment.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/main.ts',
  '/Volumes/Data/wamp/Projets/open-source/compodoc/compodoc-demo-todomvc-angular2/src/polyfills.ts' ];

let compiler = new ngdC.Compiler(
      files, {
        tsconfigDirectory: process.cwd()
      }
    );

let deps = compiler.getDependencies();

let engine = new ngdT.DotEngine({
      output: 'test',
      displayLegend: true,
      outputFormats: 'html,svg'
    });
    engine
      .generateGraph()
      .then(file => {
          console.log('end');
      });
`

i get my two files in test folder, but the svg contains only the legend.

**Comment by manekinekko** [comment]
the CLI module is in @compodoc/ngd-cli

Could you do a final test before I merge this PR and publish to npm?

**Comment by vogloblinsky** [comment]
Ok for me ;)
Don't forget compodoc account for npm publish ;)
Thanks

**Comment by manekinekko** [comment]
are you able to push this to npm? I don't remember the credentials :)

**Pull request merged**